client/App.jsx

```diff
@@ -0,0 +1,73 @@
+import React from 'react';
+import { parseRoute } from './lib';
+import Navigation from './components/Navigation';
+import NotFound from './pages/NotFound';
+import Auth from './pages/Auth';
+import OurStory from './pages/OurStory';
+import SignIn from './components/SignIn';
+import Welcome from './pages/Welcome';
+import decodeToken from './lib/decode-token';
+import Groups from './pages/Groups';
+import Quiz from './pages/Quiz';
+
+export default class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      route: parseRoute(window.location.hash)
```

Comment:
You want to make sure all properties that will be used in state have starting values here. So you should add a `user` property set to `null` and `isAuthorizing` set to `true`